Src/Support/Google.Apis/Requests/ClientServiceRequest.cs

```diff
@@ -34,12 +34,67 @@
 
 namespace Google.Apis.Requests
 {
+    /// <summary>
+    /// Represents an abstract request base class to make requests to a service.
+    /// </summary>
+    public abstract class ClientServiceRequest
+    {
+        /// <summary>Unsuccessful response handlers for this request only.</summary>
+        protected List<IHttpUnsuccessfulResponseHandler> _unsuccessfulResponseHandlers;
+        /// <summary>Exception handlers for this request only.</summary>
+        protected List<IHttpExceptionHandler> _exceptionHandlers;
+        /// <summary>Unsuccessful response handlers for this request only.</summary>
+        protected List<IHttpExecuteInterceptor> _executeInterceptors;
+
+        /// <summary>
+        /// Add an unsuccessful response handler for this request only.
+        /// </summary>
+        /// <param name="handler">The unsuccessful response handler. Must not be <c>null</c>.</param>
+        public void AddUnsuccessfulResponseHandler(IHttpUnsuccessfulResponseHandler handler)
+        {
+            handler.ThrowIfNull(nameof(handler));
+            if (_unsuccessfulResponseHandlers == null)
+            {
+                _unsuccessfulResponseHandlers = new List<IHttpUnsuccessfulResponseHandler>();
+            }
+            _unsuccessfulResponseHandlers.Add(handler);
+        }
+
+        /// <summary>
+        /// Add an exception handler for this request only.
+        /// </summary>
+        /// <param name="handler">The exception handler. Must not be <c>null</c>.</param>
+        public void AddExceptionHandler(IHttpExceptionHandler handler)
+        {
+            handler.ThrowIfNull(nameof(handler));
+            if (_exceptionHandlers == null)
+            {
+                _exceptionHandlers = new List<IHttpExceptionHandler>();
+            }
+            _exceptionHandlers.Add(handler);
+        }
+
+        /// <summary>
+        /// Add an unsuccessful response handler for this request only.
+        /// </summary>
+        /// <param name="handler">The unsuccessful response handler. Must not be <c>null</c>.</param>
+        public void AddExecuteInterceptor(IHttpExecuteInterceptor handler)
+        {
+            handler.ThrowIfNull(nameof(handler));
+            if (_executeInterceptors == null)
+            {
+                _executeInterceptors = new List<IHttpExecuteInterceptor>();
+            }
+            _executeInterceptors.Add(handler);
+        }
+    }
+
     /// <summary>
     /// Represents an abstract, strongly typed request base class to make requests to a service.
     /// Supports a strongly typed response.
     /// </summary>
     /// <typeparam name="TResponse">The type of the response object</typeparam>
-    public abstract class ClientServiceRequest<TResponse> : IClientServiceRequest<TResponse>
+    public abstract class ClientServiceRequest<TResponse> : ClientServiceRequest, IClientServiceRequest<TResponse>
```

Comment:
I'm not convinced that this change is backwardly compatible, in a binary sense. Could you just make the extension method generic in `TResponse` instead?

Reply:
Looks like I'm wrong, and this is explicitly allowed: https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md

Reply:
:)